src/components/substance/SubstanceForm.vue

```diff
@@ -124,51 +59,212 @@ import { mapGetters, mapState } from "vuex";
 
 export default {
   name: "SubstanceForm",
-  props: {
-    // Type compound being displayed.  Important for knowing which store to fetch from.
-    type: String
+  data() {
+    return {
+      changed: 0,
+      validationState: this.clearValidation(),
+      textareas: ["description", "privateQCNote", "publicQCNote"],
+      dropdowns: ["qcLevel", "source", "substanceType"],
+      labels: {
+        id: "Substance ID:",
+        preferredName: "Preferred Name:",
+        casrn: "CAS-RN:",
+        description: "Substance Description:",
+        privateQCNote: "Private QC Notes:",
+        publicQCNote: "Public QC Notes:",
+        qcLevel: "QC Level:",
+        source: "Source:",
+        substanceType: "Substance Type:"
+      }
+    };
   },
   computed: {
+    ...mapGetters("substance", ["form"]),
+    ...mapState("substance", ["detail"]),
     ...mapGetters("auth", ["isAuthenticated"]),
+    ...mapGetters("qcLevel", { qcLevelOptions: "getOptions" }),
+    ...mapGetters("source", { sourceOptions: "getOptions" }),
+    ...mapGetters("substanceType", { substanceTypeOptions: "getOptions" }),
 
-    ...mapGetters("compound/definedcompound", {
-      getDefSubstanceForm: "getSubstanceForm"
-    }),
-    ...mapGetters("compound/illdefinedcompound", {
-      getIndefSubstanceForm: "getSubstanceForm"
-    }),
-
-    ...mapState("source", { sourceList: "list" }),
-    ...mapState("substanceType", { substanceTypeList: "list" }),
-    ...mapState("qcLevel", { qcLevelList: "list" }),
-
-    form: function() {
-      // todo: don't load this from state
-      return this.$store.state.substance.form;
-    },
-    sourceOptions: function() {
-      return this.buildOptions(this.sourceList);
-    },
-    substanceTypeOptions: function() {
-      return this.buildOptions(this.substanceTypeList);
+    btnDisabled: function() {
+      return !(this.changed > 0);
     },
-    qcLevelsOptions: function() {
-      return this.buildOptions(this.qcLevelList);
+    options: function() {
+      return {
+        qcLevel: this.qcLevelOptions,
+        source: this.sourceOptions,
+        substanceType: this.substanceTypeOptions
+      };
     }
   },
-  methods: {
-    buildOptions: function(list) {
-      let item;
-      let options = [];
-      for (item of list)
-        options.push({ value: item.id, text: item.attributes.label });
-      return options;
+  watch: {
+    "form.id": function() {
+      this.validationState = this.clearValidation();
+      this.changed = 0;
     }
   },
-  mounted() {
-    this.$store.dispatch("source/getList");
-    this.$store.dispatch("qcLevel/getList");
-    this.$store.dispatch("substanceType/getList");
+  methods: {
+    editable(fld) {
+      return fld === "id" ? true : !this.isAuthenticated;
+    },
+    markChanged() {
+      this.changed++;
+    },
```

Comment:
it'll work the way you suggest if we use the event `@input` but I was having another issue with that, let me look into it some more, I'd like the `changed` prop to hold an array of the field names that are changed, I could use that in updatePayload to filter more simply

Reply:
oh, the reason I used `@change` was because the dropdowns were firing the event on `clearForm` I think if I just use `@change` in the dropdowns and `@input` in the other inputs we'll get what we want here

Reply:
As long as there was a valid reason for why you couldn't approach it to be more responsive, I'm fine.  This would be a small polish thing.

That said could you use `@input` for the text area / input fields and `@change` for the drop down?